pkg/dao/rpms.go

```diff
@@ -620,3 +619,73 @@ func (r *rpmDaoImpl) ListSnapshotErrata(ctx context.Context, orgId string, snaps
 
 	return response, total, nil
 }
+
+func (r *rpmDaoImpl) ListTemplateRpms(ctx context.Context, orgId string, templateUUID string, search string, pageOpts api.PaginationData) ([]api.SnapshotRpm, int, error) {
+	response := []api.SnapshotRpm{}
+	pulpHrefs := []string{}
+
+	snapshotUuids, err := FetchSnapshotsForTemplate(ctx, r.db, orgId, templateUUID)
+	if err != nil {
+		return response, 0, err
+	}
+
+	res := readableSnapshots(r.db.WithContext(ctx), orgId).Where("snapshots.UUID in ?", UuidifyStrings(snapshotUuids)).Pluck("version_href", &pulpHrefs)
+	if res.Error != nil {
+		return response, 0, fmt.Errorf("failed to query the db for snapshots: %w", res.Error)
+	}
+	if config.Tang == nil {
+		return response, 0, fmt.Errorf("no tang configuration present")
+	}
+
+	if len(pulpHrefs) == 0 {
+		return response, 0, nil
+	}
+
+	pkgs, total, err := (*config.Tang).RpmRepositoryVersionPackageList(ctx, pulpHrefs, tangy.RpmListFilters{Name: search}, tangy.PageOptions{
+		Offset: pageOpts.Offset,
+		Limit:  pageOpts.Limit,
+	})
+
+	if err != nil {
+		return response, 0, fmt.Errorf("error querying packages in templates: %w", err)
+	}
+	for _, pkg := range pkgs {
+		response = append(response, api.SnapshotRpm{
+			Name:    pkg.Name,
+			Arch:    pkg.Arch,
+			Version: pkg.Version,
+			Release: pkg.Release,
+			Epoch:   pkg.Epoch,
+			Summary: pkg.Summary,
+		})
+	}
+	return response, total, nil
+}
+
+func FetchSnapshotsForTemplate(ctx context.Context, db *gorm.DB, orgId string, templateUUID string) ([]string, error) {
```

Comment:
I would just make this a method on the struct like ListTemplateRpms.  it doesn't have to be declared in the interface and could even be private (by making it start with a lowercase letter)